os/include/net/lwip/apps/snmp_core.h

```diff
@@ -0,0 +1,364 @@
+#define SNMP_ASN1_TYPE_COUNTER64      (SNMP_ASN1_CLASS_APPLICATION | SNMP_ASN1_CONTENTTYPE_PRIMITIVE | SNMP_ASN1_APPLICATION_COUNTER64)
+
+#define SNMP_VARBIND_EXCEPTION_OFFSET 0xF0
+#define SNMP_VARBIND_EXCEPTION_MASK   0x0F
+
+/** error codes predefined by SNMP prot. */
+typedef enum {
+	SNMP_ERR_NOERROR = 0,
+/*
+outdated v1 error codes. do not use anmore!
+#define SNMP_ERR_NOSUCHNAME 2  use SNMP_ERR_NOSUCHINSTANCE instead
+#define SNMP_ERR_BADVALUE   3  use SNMP_ERR_WRONGTYPE,SNMP_ERR_WRONGLENGTH,SNMP_ERR_WRONGENCODING or SNMP_ERR_WRONGVALUE instead
+#define SNMP_ERR_READONLY   4  use SNMP_ERR_NOTWRITABLE instead
+*/
+	SNMP_ERR_GENERROR = 5,
+	SNMP_ERR_NOACCESS = 6,
+	SNMP_ERR_WRONGTYPE = 7,
+	SNMP_ERR_WRONGLENGTH = 8,
+	SNMP_ERR_WRONGENCODING = 9,
+	SNMP_ERR_WRONGVALUE = 10,
+	SNMP_ERR_NOCREATION = 11,
+	SNMP_ERR_INCONSISTENTVALUE = 12,
+	SNMP_ERR_RESOURCEUNAVAILABLE = 13,
+	SNMP_ERR_COMMITFAILED = 14,
+	SNMP_ERR_UNDOFAILED = 15,
+	SNMP_ERR_NOTWRITABLE = 17,
+	SNMP_ERR_INCONSISTENTNAME = 18,
+
+	SNMP_ERR_NOSUCHINSTANCE = SNMP_VARBIND_EXCEPTION_OFFSET + SNMP_ASN1_CONTEXT_VARBIND_NO_SUCH_INSTANCE
+} snmp_err_t;
+
+/** internal object identifier representation */
+struct snmp_obj_id {
+	u8_t len;
+	u32_t id[SNMP_MAX_OBJ_ID_LEN];
+};
+
+struct snmp_obj_id_const_ref {
+	u8_t len;
+	const u32_t *id;
+};
+
+extern const struct snmp_obj_id_const_ref snmp_zero_dot_zero;	/* administrative identifier from SNMPv2-SMI */
+
+/** SNMP variant value, used as reference in struct snmp_node_instance and table implementation */
+union snmp_variant_value {
+	void *ptr;
+	const void *const_ptr;
+	u32_t u32;
+	s32_t s32;
+};
+
+/**
+SNMP MIB node types
+ tree node is the only node the stack can process in order to walk the tree,
+ all other nodes are assumed to be leaf nodes.
+ This cannot be an enum because users may want to define their own node types.
+*/
+#define SNMP_NODE_TREE         0x00
+/* predefined leaf node types */
+#define SNMP_NODE_SCALAR       0x01
+#define SNMP_NODE_SCALAR_ARRAY 0x02
+#define SNMP_NODE_TABLE        0x03
+#define SNMP_NODE_THREADSYNC   0x04
+
+/** node "base class" layout, the mandatory fields for a node  */
+struct snmp_node {
+	/** one out of SNMP_NODE_TREE or any leaf node type (like SNMP_NODE_SCALAR) */
+	u8_t node_type;
+	/** the number assigned to this node which used as part of the full OID */
+	u32_t oid;
+};
+
+/** SNMP node instance access types */
+typedef enum {
+	SNMP_NODE_INSTANCE_ACCESS_READ = 1,
+	SNMP_NODE_INSTANCE_ACCESS_WRITE = 2,
+	SNMP_NODE_INSTANCE_READ_ONLY = SNMP_NODE_INSTANCE_ACCESS_READ,
+	SNMP_NODE_INSTANCE_READ_WRITE = (SNMP_NODE_INSTANCE_ACCESS_READ | SNMP_NODE_INSTANCE_ACCESS_WRITE),
+	SNMP_NODE_INSTANCE_WRITE_ONLY = SNMP_NODE_INSTANCE_ACCESS_WRITE,
+	SNMP_NODE_INSTANCE_NOT_ACCESSIBLE = 0
+} snmp_access_t;
+
+struct snmp_node_instance;
+
+typedef s16_t(*node_instance_get_value_method) (struct snmp_node_instance *, void *);
```

Comment:
for typdef of fucntion prototype, ```return_type (*function_name)(argument);``` is right.
1. add a spave between return tye and function name
2. remove a space between function name and argument

Reply:
Fixed